casual/casual-caller-http-client/src/main/java/se/laz/casual/http/HttpClient.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2024, The casual project. All rights reserved.
+ *
+ * This software is licensed under the MIT license, https://opensource.org/licenses/MIT
+ */
+package se.laz.casual.http;
+
+import jakarta.transaction.Transactional;
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.ClientBuilder;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import se.laz.casual.api.buffer.CasualBuffer;
+import se.laz.casual.api.buffer.CasualBufferType;
+import se.laz.casual.api.buffer.ServiceReturn;
+import se.laz.casual.api.buffer.type.ServiceBuffer;
+import se.laz.casual.api.flags.ErrorState;
+import se.laz.casual.api.flags.ServiceReturnState;
+
+import java.io.Closeable;
+import java.net.URI;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.logging.Logger;
+
+import static jakarta.ws.rs.client.Entity.entity;
+
+public class HttpClient implements Closeable
+{
+    private static final Logger LOG = Logger.getLogger(HttpClient.class.getName());
+    private final Client client;
+    public HttpClient()
+    {
+        // CDI public constructor
+        this.client = ClientBuilder.newClient();
+    }
+
+    public static HttpClient of()
+    {
+        return new HttpClient();
+    }
+
+    @Transactional(Transactional.TxType.NOT_SUPPORTED)
+    public ServiceReturn<CasualBuffer> request(URI uri, CasualBuffer buffer)
+    {
+        Objects.requireNonNull(uri, "uri can not be null");
+        Objects.requireNonNull(buffer, "buffer can not be null");
+        MediaType mediaType = CasualBufferTypeConverter.convert(CasualBufferType.unmarshall(buffer.getType()));
+        LOG.finest(() -> "issuing http request to " + uri);
+        Response response = client.target(uri).request(mediaType).post(entity(buffer.getBytes().get(0), mediaType));
+        ErrorState errorState = ResponseStatusConverter.convert(response.getStatusInfo().toEnum());
+        if (errorState != ErrorState.OK)
+        {
+            return errorResponse(errorState, response);
+        }
+        Optional<CasualBuffer> responseBuffer = ResponseConverter.convert(response);
```

Comment:
Why use `Optional` here? It is not an optional response, there is always a response buffer but sometimes it is empty.

If you have `ResponseConverter` return a `CasualBuffer` instead and when it was returning `Optional.empty()` before it will now return `ServiceBuffer.empty()`.
You can then remove the duplication of the `new ServiceReturn` in the following lines.
You have the exact same thing on `errorResponse` too.

Reply:
Sure